File: Backend/PlanItRight/EventManagementService/src/main/java/com/PlanItRight/EventManagementService/Model/Task.java
package com.PlanItRight.EventManagementService.model;



import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Index;

import java.sql.Time;
import java.util.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private Date dueDate;
    private Time dueTime;
    private String status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id", nullable = false)
    @JsonIgnore
    private com.PlanItRight.EventManagementService.model.Event event;
}

File: Frontend/PlanItRight/src/app/task.ts
export class Task {
    id: number;
    name: string;
    description: string;
    dueDate: Date;
    status: string;

    constructor(id: number, name: string, description: string, dueDate: Date, status: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.dueDate = dueDate;
        this.status = status;
    }
}
